app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -16,8 +16,9 @@ def next_setup_path
           second_mfa_reminder_conversion: user_session.delete(:second_mfa_reminder_conversion),
           success: true,
         )
+        user_session.delete(:mfa_selections)
+        return sign_up_completed_path
```

Comment:
Stylistically I think we could flatten this out to avoid the nested `if` since we're no longer doing anything outside meaningful outside the `if` block, relying on implicit `nil` return.

```rb
  def next_setup_path
    if suggest_second_mfa?
      auth_method_confirmation_url
    elsif next_setup_choice
      confirmation_path
    elsif user_session[:mfa_selections]
      analytics.user_registration_mfa_setup_complete(
        mfa_method_counts: mfa_context.enabled_two_factor_configuration_counts_hash,
        in_account_creation_flow: user_session[:in_account_creation_flow] || false,
        enabled_mfa_methods_count: mfa_context.enabled_mfa_methods_count,
        pii_like_keypaths: [[:mfa_method_counts, :phone]],
        second_mfa_reminder_conversion: user_session.delete(:second_mfa_reminder_conversion),
        success: true,
      )
      user_session.delete(:mfa_selections)
      sign_up_completed_path
    end
  end
```